### ISSUE TITLE
[UE4] Request: set up the atlas and skeleton data assets so they can be procedurally assigned

### ISSUE THREAD

**Issue by antidamage** [EsotericSoftware/spine-runtimes#1176]
I had a crack at this myself but I'm not brilliant in the UE framework and only got as far as adding build errors.

Here's a screenshot that kind of outlines the problem: https://i.imgur.com/pVziCJi.png

Right now the skeleton and atlas data aren't set up right in the framework and they can't be promoted to variables, so they can't be assigned via variables from other objects.

In the project above I have a virtual book object with many pages, but because of the current limitations I have to create a new actor blueprint with spine components for every single spine character I want to add to the book, spawn that and then attach it instead of using the components directly. Because it's all custom hand-drawn art there's at least one character per page, so it's becoming tedious.

This has ramifications for things like the construction scripts in the editor which can't spawn actors, so I'm finding it difficult to set up clickable hotspots because the spine actors aren't visible until you hit play. I need them to be procedurally created components.

TL;DR: need to be able to promote the asset reference to a variable so it can be used during construction to create and assign spine characters procedurally.

**Comment by badlogic** [comment]
Thanks for this. Sadly, we aren't UE4 experts either, and the docs are a bit lacking when it comes to customizing things.

**Comment by AndreaDev237** [comment]
Hi Everyone,
to achive what you want you have to modify the **`SpineAtlasAsset.h`** and **`SpineSkeletonDataAsset.h`**

add **`BlueprintType`** paramater to **`UCLASS`** Macro 

in this way you tell to engine this type can be referenced in a BlueprintType Var 

![image](https://user-images.githubusercontent.com/2900985/45943893-7d482c80-bfe8-11e8-81f6-c0b40113ee8e.png)

![image](https://user-images.githubusercontent.com/2900985/45943971-c5ffe580-bfe8-11e8-8f68-2078892ad681.png)

![image](https://user-images.githubusercontent.com/2900985/45944206-969da880-bfe9-11e8-9fd0-c632e1a6e9d0.png)

However in the 3.7-beta-cpp Branch, badlogic has already expose that type to Blueprint so i think you have to update the runtime with last version or manually modify code with your current version

**Comment by badlogic** [comment]
@ps2invader ahh, I remember that change now. To many runtimes to shuffle, thanks for reminding me! @antidamage I assume this resolves this issue.
